launcher/ui/dialogs/MSALoginDialog.cpp

```diff
@@ -161,22 +97,38 @@ void MSALoginDialog::onTaskSucceeded()
 
 void MSALoginDialog::onTaskStatus(const QString& status)
 {
-    ui->label->setText(status);
-}
-
-void MSALoginDialog::onTaskProgress(qint64 current, qint64 total)
-{
-    ui->progressBar->setMaximum(total);
-    ui->progressBar->setValue(current);
+    ui->message->setText(status);
+    ui->cancel->setEnabled(false);
+    ui->link->setVisible(false);
+    ui->copy->setVisible(false);
 }
 
 // Public interface
 MinecraftAccountPtr MSALoginDialog::newAccount(QWidget* parent, QString msg)
 {
     MSALoginDialog dlg(parent);
-    dlg.ui->label->setText(msg);
+    dlg.ui->message->setText(msg);
     if (dlg.exec() == QDialog::Accepted) {
         return dlg.m_account;
     }
     return nullptr;
 }
+
+void MSALoginDialog::authorizeWithBrowser(const QUrl& url)
+{
+    ui->cancel->setEnabled(true);
+    ui->link->setVisible(true);
+    ui->copy->setVisible(true);
+    DesktopServices::openUrl(url);
+    ui->link->setText(url.toDisplayString());
+    ui->message->setText(
+        tr("Browser opened to complete the login process."
+           "<br /><br />"
+           "If your browser hasn't opened, please manually open the bellow link in your browser:"));
```

Comment:
```suggestion
           "If your browser hasn't opened, please manually open the below link in your browser:"));
```
maybe even swap below and link, I'm not that great at English grammar either though, its not my first language x3